vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/JcrPackageRegistry.java

```diff
@@ -241,6 +249,11 @@ public Node getPrimaryPackageRoot(boolean autoCreate) throws RepositoryException
     public RegisteredPackage open(@Nonnull PackageId id) throws IOException {
         try {
             Node node = getPackageNode(id);
+            if (node == null && baseRegistry != null) {
+                if (baseRegistry.contains(id)) {
```

Comment:
can't this be simplified to just:

```suggestion
                return baseRegistry.open(id)
```

or does it bring additional value to check `contains()` first?